frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditTaskRecommendedItems/NewNameRecommendation.tsx

```diff
@@ -0,0 +1,62 @@
+import { useValidateBpmnTaskId } from '../../../../hooks/useValidateBpmnId';
+import { useBpmnContext } from '../../../../contexts/BpmnContext';
+import {
+  StudioIconTextfield,
+  StudioRecommendedNextAction,
+  useStudioRecommendedNextActionContext,
+} from '@studio/components';
+import { KeyVerticalIcon } from '@studio/icons';
+import type Modeling from 'bpmn-js/lib/features/modeling/Modeling';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export const NewNameRecommendation = (): React.ReactElement => {
+  const { bpmnDetails, setBpmnDetails, modelerRef } = useBpmnContext();
+  const modelerInstance = modelerRef.current;
+  const modeling: Modeling = modelerInstance.get('modeling');
+  const { t } = useTranslation();
+  const { validateBpmnTaskId } = useValidateBpmnTaskId();
+  const { removeAction } = useStudioRecommendedNextActionContext();
+
+  const [newName, setNewName] = useState('');
+  const [newNameError, setNewNameError] = useState('');
+
+  const saveNewName = () => {
+    removeAction(bpmnDetails.element.id);
+    modeling.updateProperties(bpmnDetails.element, {
+      id: newName,
+    });
+    setBpmnDetails({
+      ...bpmnDetails,
+      id: newName,
+    });
+  };
```

Comment:
Ref. previous comment, we do have abstraction to simplify the usage of the modelling tools from bpmnjs package.

```suggestion
    studioModeler.updateElementProperties({ id: newName });
    setBpmnDetails({
      ...bpmnDetails,
      id: newName,
    });
  };
```